homeassistant/components/moon/sensor.py

```diff
@@ -13,7 +13,25 @@
 
 DEFAULT_NAME = "Moon"
 
-ICON = "mdi:brightness-3"
+STATE_NEW_MOON = "new_moon"
+STATE_WAXING_CRESCENT = "waxing_crescent"
+STATE_FIRST_QUARTER = "first_quarter"
+STATE_WAXING_GIBBOUS = "waxing_gibbous"
+STATE_FULL_MOON = "full_moon"
+STATE_WANING_GIBBOUS = "waning_gibbous"
+STATE_LAST_QUARTER = "last_quarter"
+STATE_WANING_CRESCENT = "waning_crescent"
```

Comment:
Please replace the strings for `self.state` with the constants.

Reply:
sorry, but I dont understand what this means. Please give me an example, and I will follow suit with the rest. thanks.

Reply:
I think he means in the state function where it returns for example

```python 
return "new_moon"
```

you should change this for 
```python 
return STATE_NEW_MOON
```

and so on for all the other states.

Reply:
I see, thanks, so like this:
```
    @property
    def state(self):
        """Return the state of the device."""
        if self._state == 0:
            return "STATE_NEW_MOON"
        if self._state < 7:
            return "STATE_WAXING_CRESCENT"
        if self._state == 7:
            return "STATE_FIRST_QUARTER"
        if self._state < 14:
            return "STATE_WAXING_GIBBOUS"
        if self._state == 14:
            return "STATE_FULL_MOON"
        if self._state < 21:
            return "STATE_WANING_GIBBOUS"
        if self._state == 21:
            return "STATE_LAST_QUARTER"
        return "STATE_WANING_CRESCENT"
```

?

Reply:
done

Reply:
without quotes

Reply:
removed....